jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/DiagnosticUtils.java

```diff
@@ -272,17 +284,50 @@ private static DiagnosticProperties doDump(String diagnosticCommand) {
 				}
 			}
 			if (!dumpType.isEmpty()) {
-				String request = dumpType;
-				if (parts.length == 2) {
-					String fileDirective = ("system".equals(dumpType) && IPC.isZOS) ? ":dsn=" : ":file="; //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
-					request += fileDirective + parts[1];
+				StringBuilder request = new StringBuilder();
+				request.append(dumpType);
+				String filePath = null;
+				boolean foundRequest = false;
+				boolean heapDump = "heap".equals(dumpType); //$NON-NLS-1$
+				boolean systemDump = "system".equals(dumpType); //$NON-NLS-1$
+				String separator = ":"; //$NON-NLS-1$
```

Comment:
`:` is only appropriate before the first option; subsequent options are separated by `,`.